src/lib/matcher.rs

```diff
@@ -516,11 +516,13 @@ pub mod tests {
             let opts = MatcherOpts {
                 invert_match,
                 reverse_complement,
-                color: false,
             };
             let matcher = FixedStringSetMatcher::new(patterns.iter(), opts);
-            let mut read_record = write_owned_record(seq);
-            let result = matcher.read_match(&mut read_record);
+            let qual = (0..seq.len()).map(|_| "X").collect::<String>();
+            let record = format!("@id\n{seq}\n+\n{qual}\n");
+            let mut reader = seq_io::fastq::Reader::new(record.as_bytes());
+            let read_record = reader.next().unwrap().unwrap();
+            let result = matcher.read_match(&read_record);
```

Comment:
I couldn't figure out how to refactor this into their own function due to lifetimes.  Deferring since it's test code.